serviceregistry/src/main/java/eu/arrowhead/serviceregistry/jpa/service/SystemDbService.java

```diff
@@ -317,10 +317,6 @@ public Page<Triple<System, List<SystemAddress>, Entry<Device, List<DeviceAddress
 						}
 
 						// device names
-						if (!Utilities.isEmpty(deviceNames) && deviceSystemConnectorRepo.findBySystem(system).isPresent()
-								&& !deviceNames.contains(deviceSystemConnectorRepo.findBySystem(system).get().getDevice().getName())) {
-							continue;
-						}
 
 						if (!Utilities.isEmpty(deviceNames)) {
 							final Optional<DeviceSystemConnector> connection = deviceSystemConnectorRepo.findBySystem(system);
```

Comment:
We found a bug here. If a system has no device this condition does not filter it, but it should.

We think this is the right condition:
`connection.isEmpty() || !deviceNames.contains(connection.get().getDevice().getName())`

I suppose this will changes some tests too. Sorry.